src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -283,6 +295,86 @@ public void InitializeService(ServiceHost serviceHost, SqlToolsContext context)
 
         #region Request Handlers
 
+        /// <summary>
+        /// Completion extension load request callback
+        /// </summary>
+        /// <param name="param"></param>
+        /// <param name="requestContext"></param>
+        /// <returns></returns>
+        internal async Task HandleCompletionExtLoadRequest(CompletionExtensionParams param, RequestContext<bool> requestContext)
+        {
+            try
+            {
+                if (!CheckExtAssemblyToBeLoaded(param.AssemblyPath))
+                {
+                    await requestContext.SendError(string.Format("Skip loading {0} because it's already loaded once or doesn't exist", param.AssemblyPath));
+                    return;
+                }
+                //register the new assembly
+                var serviceProvider = (ExtensionServiceProvider)ServiceHostInstance.ServiceProvider;
+                var assembly = AssemblyLoadContext.Default.LoadFromAssemblyPath(param.AssemblyPath);
+                var assemblies = new Assembly[] { assembly };
+                serviceProvider.AddAssembliesToConfiguration(assemblies);
+                foreach (var ext in serviceProvider.GetServices<ICompletionExtension>())
+                {
+                    var cancellationTokenSource = new CancellationTokenSource(ExtensionLoadingTimeout);
+                    var cancellationToken = cancellationTokenSource.Token;
+                    string extName = ext.Name;
+                    string extTypeName = ext.GetType().FullName;
+                    if (extTypeName != param.TypeName)
+                    {
+                        continue;
+                    }
+                    await ext.Initialize(param.Properties, cancellationToken).WithTimeout(ExtensionLoadingTimeout);
+                    cancellationTokenSource.Dispose();
+                    if (!string.IsNullOrEmpty(extName))
+                    {
+                        completionExtensions[extName] = ext;
+                        await requestContext.SendResult(true);
+                        return;
+                    }
+                    else
+                    {
+                        await requestContext.SendError(string.Format("Skip loading an unnamed completion extension from {0}", param.AssemblyPath));
+                        return;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                await requestContext.SendError(ex.Message);
+                return;
+            }
+        }
+
+        /// <summary>
+        /// Check a particular assembly needs to be loaded or not
```

Comment:
>Check a particular assembly needs to be loaded or not [](start = 12, length = 53)

Check whether a particular assembly should be reloaded based on whether it's been updated since it was last loaded.